src/System.Management.Automation/utils/tracing/SysLogProvider.cs

```diff
@@ -0,0 +1,543 @@
+                args = new object[] {eventId};
+            }
+
+            string resourceValue = GetResourceString(resourceName);
+            if (parameterCount > 0)
+            {
+                sb.AppendFormat(resourceValue, args);
+            }
+            else
+            {
+                sb.Append(resourceValue);
+            }
+        }
+
+        #region logging
+
+        // maps a LogLevel to an associated SysLogPriority
+        static NativeMethods.SysLogPriority[] _levels =
+        {
+            NativeMethods.SysLogPriority.Info,
+            NativeMethods.SysLogPriority.Critical,
+            NativeMethods.SysLogPriority.Error,
+            NativeMethods.SysLogPriority.Warning,
+            NativeMethods.SysLogPriority.Info,
+            NativeMethods.SysLogPriority.Info
+        };
+
+        /// <summary>
+        /// Logs a activity transfer
+        /// </summary>
+        /// <param name="parentActivityId">The parent activity id.</param>
+        public void LogTransfer(Guid parentActivityId)
+        {
+            // NOTE: always log
+            int threadId = Thread.CurrentThread.ManagedThreadId;
+            string message = string.Format(CultureInfo.InvariantCulture,
+                                           "({0:X}:{1:X}:{2:X}) [Transfer]:{3} {4}",
+                                           _commitId, threadId, PSChannel.Operational,
+                                           parentActivityId.ToString("B"),
+                                           Activity.ToString("B"));
+
+            NativeMethods.SysLog(NativeMethods.SysLogPriority.Info, message);
+        }
+
+        /// <summary>
+        /// Logs the activity identifier for the current thread.
+        /// </summary>
+        /// <param name="activity">The Guid activity identifier</param>
+        public void SetActivity(Guid activity)
+        {
+            int threadId = Thread.CurrentThread.ManagedThreadId;
+            Activity = activity;
+
+            // NOTE: always log
+            string message = string.Format(CultureInfo.InvariantCulture,
+                                           "({0:X}:{1:X}:{2:X}) [Activity] {3}",
+                                           _commitId, threadId, PSChannel.Operational, activity.ToString("B"));
+            NativeMethods.SysLog(NativeMethods.SysLogPriority.Info, message);
+        }
+
+        /// <summary>
+        /// Logs a message that provides a correlation between the full git commit id string and a hash code used for subsequent log entries.
+        /// </summary>
+        static void LogCommit()
+        {
+            int threadId = Thread.CurrentThread.ManagedThreadId;
+            // NOTE: always log
+            string message = string.Format(CultureInfo.InvariantCulture,
+                                           "({0:X}:{1:X}:{2:X}) [GitCommitId] {3} Hash: {0:X}",
+                                           _commitId, threadId, PSChannel.Operational, PSVersionInfo.GitCommitId);
```

Comment:
I see we log `PSVersionInfo.GitCommitId`. 
1. We should replace `Hash: {0:X}` with `GitCommitId: {0:X}`
2. We can and should use first 7 chars from Git SHA in _commitId - if we log full GitCommitId string we can always resolve a conflict of coincidence if it appears. (Although this coincidence is unlikely)

Reply:
Won't fix. the point of this is to NOT have every log entry taken up by the full commit id.
As far as the first 7 chars, I don't know what collision % that will cause and haven't seen any data indicating it.  If I change this, I'm more inclined to use a modified String.GetHashCode that doesn't change across builds or processes.

Reply:
I think we can use a short SHA:
1. We always write this whole GitCommitId. So if MSFT support receives the log, it can always see the exact version.
2. I think we're talking about final releases. We will have a small number of these versions. Collision probability is zero.
3. Conflicts are possible with a very large number of commits. I believe the repo isn't large.  [See](https://stackoverflow.com/questions/18134627/how-much-of-a-git-sha-is-generally-considered-necessary-to-uniquely-identify-a). We can easy export SHAs from the repo and check collisions.

Reply:
Won't fix.

This discussion turns out to be moot.  Official builds use the release tag for the GitCommitId; other builds appear to have the SHA-1 has appended to the build.

I'm going to leave the logic as is but will consider a repeatable hash code.

Reply:
@dantraMSFT @SteveL-MSFT @iSazonov We need a conclusion about which hash to use for represent a build. Or we can accept the PR as is and open an issue to continue the discussion about which hash code to use.

Reply:
> Official builds use the release tag for the GitCommitId; other builds appear to have the SHA-1 has appended to the build.

Official build also has the SHA-1 commit id embedded with the assembly. You can get it by
```
> $s = [psobject].Assembly.GetCustomAttributes([System.Reflection.AssemblyInformationalVersionAttribute], $false)
> $s.InformationalVersion
6.0.0-beta.7 SHA: d2fe36fb4137c2a7d6dbc0df22245d14256f3a74
```

> will consider a repeatable hash code.

This also sounds good to me. To me, a consistent hash across processes would be good enough.

@dantraMSFT Can you please open an issue tracking the "repeatable has code" change that you are thinking about?

Reply:
https://github.com/PowerShell/PowerShell/issues/5250

Reply:
For most users, `$psversiontable.gitcommitid` is the release tag which is short and descriptive and I think perfectly fine for this purpose which is to enable filtering of the traces if you have 6.0.0 and 6.1.0 running side-by-side.  For developers running private builds where the gitcommitid has the long hash.  I don't see it as a big issue to log the whole thing.  I'm also fine if we decide to just log the first 7 characters of the hash as in most cases, I think it's unlikely for there to be a collision and I also expect most developers who care about this to also filter by datetime so even if there is a collision, they are probably not running two private builds concurrently that also collide on the first 7 characters of the hash.  I think we're spending too much time over-engineering this particular issue.

Reply:
Agreed. I'll just use the GitCommitId property with no parsing and see what pushback we get.

Reply:
Fixed